Sources/SafeDICore/Generators/DependencyTreeGenerator.swift

```diff
@@ -354,11 +353,9 @@ public final class DependencyTreeGenerator {
                     validateReceivedProperties(
                         on: childScope,
                         receivableProperties: receivableProperties
-                            .union(scope.properties)
-                            .removing(childProperty),
```

Comment:
this `.removing(childProperty)` used to be super important, since when SafeDI was originally written our cycle detection was effectively in `func validateReceivedProperties` – the error messages in this function were not great at describing cycles, but it's how we accomplished the task. Now we've moved cycle detection outside of this method and into the `createScopeGenerator`.

Reply:
I have an inkling that we _should_ move all cycle detection to `validateReceivedProperties` and rename that method `validateTreeIsFulfillable`. The way we've split the cycle detection across a few places smells to me. But that can be a separate PR. Our test suite is pretty solid so I'm not worried about regressions at the moment.

Reply:
alright I took a pass here. `validateReceivedProperties` now has more of the checking, but not all of it. Pulling the received property checking our of the scope generator is not easy and not high value.